enhancements/baremetal/baremetal-keepalived-unicast.md

```diff
@@ -0,0 +1,165 @@
+
+##### Control plane nodes
+
+The [keepalived pod](https://github.com/openshift/machine-config-operator/blob/4c04978faf8ce36af046e18c7722d03f6e60133e/templates/common/baremetal/files/baremetal-keepalived.yaml) in control plane nodes already includes keepalived and keepalived-monitor containers. 
+To support unicast mode, the keepalived-monitor should be updated to include the [bootstrap node IP address](https://github.com/openshift/baremetal-runtimecfg/blob/8242fd84d6c6d267d2d5551abf8e4f778b4f0615/pkg/monitor/dynkeepalived.go#L30) and [IP addresses of control plane nodes](https://github.com/openshift/baremetal-runtimecfg/blob/8242fd84d6c6d267d2d5551abf8e4f778b4f0615/pkg/config/node.go#L336-#L380).
+
+To avoid a circular dependency (keepalived--> api-vip--> keepalived) the control plane nodes details retrieved from localhost:kube-apiserver and not api-vip:kube-apiserver.
+
+#### ingress-vip implementation
+
+Add unicast support for the `ingress-vip` is simpler than the `api-vip`.
+
+The following steps should be done to enable unicast for `ingress-vip`:
+1. Update the keepalived conf template files for both [control plane nodes](https://github.com/openshift/machine-config-operator/blob/4c04978faf8ce36af046e18c7722d03f6e60133e/templates/master/00-master/baremetal/files/baremetal-keepalived-keepalived.yaml#L49-#L73) and [compute nodes](https://github.com/openshift/machine-config-operator/blob/4c04978faf8ce36af046e18c7722d03f6e60133e/templates/worker/00-worker/baremetal/files/baremetal-keepalived-keepalived.yaml) to include the unicast details.
+2. The keepalived-monitor container should be updated to [retrieve](https://github.com/openshift/baremetal-runtimecfg/blob/8242fd84d6c6d267d2d5551abf8e4f778b4f0615/pkg/config/node.go#L219-#L244) the nodes (both compute and control) IP addresses from the kube-apiserver and render them to keepalived config.
+
+### Risks and Mitigations
+
+It is important to not destabilize our existing installation method, that uses
+multicast successfully. Full parity is expected before landing.
+
+## Design Details
+
+### Test Plan
+
+Because of the fundamental inclusion of this in Baremetal IPI installations,
+any breakage by CI before landing can probably be assumed to be caused by
+this work.
+
+## Implementation History
+
+A PR with implementation in baremetal-runtimecfg can be found here:
+
+* https://github.com/openshift/baremetal-runtimecfg/pull/65
+
+The Machine Config operator portion can be found here:
+
+* https://github.com/openshift/machine-config-operator/pull/1768
+
+## Drawbacks
+
+* Multicast is a simpler configuration with fewer moving parts. However, not every
+  environment allows multicast, whereas all environments would allow unicast.
+
+## Alternatives
+
+An alternative would be to absolutely require multicast for installation and
+operation.
```

Comment:
We could also have a sort of "stretch goal" where the runtimecfg can check for a CRD in order to switch to VRRP mode for those environments that want to use it after installation/clustering.

Reply:
Well, that means we have two different ways to control keepalived mode, which might lead to conflicts. I would prefer to have a single way to control Unicast/Multicast.

Is it possible to update the Keepalived mode by patching the MCO manifests?